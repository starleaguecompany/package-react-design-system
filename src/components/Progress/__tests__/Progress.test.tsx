import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { Progress } from '../../../index'

describe('Progress', () => {
  test('should render correctly with no props', () => {
    render(<Progress />)

    expect(screen.getByTestId('Progress')).toMatchSnapshot()
    expect(screen.getByTestId('Progress')).toHaveClass('container', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<Progress id="test-id" className="test-class" />)

    expect(screen.getByTestId('Progress')).toMatchSnapshot()
    expect(screen.getByTestId('Progress')).toHaveClass('test-class')
    expect(screen.getByTestId('Progress')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with percent prop', () => {
    const { container } = render(<Progress percent={50} />)
    const progressBar = container.querySelector('.progress') as HTMLElement

    expect(screen.getByTestId('Progress')).toMatchSnapshot()
    expect(screen.getByTestId('Progress')).toContainElement(progressBar)
    expect(progressBar).toHaveStyle('width: 50%')
  })
})
