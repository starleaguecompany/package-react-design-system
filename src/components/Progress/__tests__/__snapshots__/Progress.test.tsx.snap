// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="Progress"
  id="test-id"
>
  <div
    class="inner"
  >
    <div
      class="progress"
      style="width: 0%;"
    />
  </div>
</div>
`;

exports[`Progress should render correctly with no props 1`] = `
<div
  class="container"
  data-qa="Progress"
>
  <div
    class="inner"
  >
    <div
      class="progress"
      style="width: 0%;"
    />
  </div>
</div>
`;

exports[`Progress should render correctly with percent prop 1`] = `
<div
  class="container"
  data-qa="Progress"
>
  <div
    class="inner"
  >
    <div
      class="progress"
      style="width: 50%;"
    />
  </div>
</div>
`;
